extensions/podman/src/warnings.ts

```diff
@@ -67,19 +67,23 @@ export async function isDisguisedPodman(): Promise<boolean> {
   };
 
   return new Promise<boolean>(resolve => {
-    const req = http.get(podmanPingUrl, res => {
-      res.on('data', () => {
-        // do nothing
-      });
+    const req = http
+      .get(podmanPingUrl, res => {
```

Comment:
I would probably use the timeout option of the get method 
https://nodejs.org/api/http.html#httprequestoptions-callback

For example https://github.com/containers/podman-desktop/blob/main/packages/main/src/plugin/extensions-catalog/extensions-catalog.ts#L142-148


I would also file an issue about limiting the time the activate method of an extension can take
For example activation of an extension should not take more than 5s.

Reply:
> I would probably use the timeout option of the get method https://nodejs.org/api/http.html#httprequestoptions-callback
> 
> For example https://github.com/containers/podman-desktop/blob/main/packages/main/src/plugin/extensions-catalog/extensions-catalog.ts#L142-148

I had tried adding `timeout: 5000` in podmanPingUrl and listening for timeout events within the get() first, but I couldn't get it to actually timeout; adding the separate setTimeout() worked. When googling all the examples (even after the timeout property was 'recently' added) were doing the same thing.

Seems I'm not the only one - sporadic reports here, mostly for the timeout option: https://github.com/nodejs/node/issues/23282

> I would also file an issue about limiting the time the activate method of an extension can take For example activation of an extension should not take more than 5s.

#3280 opened.